lib/vet360/stats.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Vet360
+  class Stats
+    FINAL_SUCCESS = %w[COMPLETED_SUCCESS COMPLETED_NO_CHANGES_DETECTED].freeze
+    FINAL_FAILURE = %w[REJECTED COMPLETED_FAILURE].freeze
+
+    class << self
+      # Parses our exceptions file and returns all of the Vet360 exception keys.
+      #
+      # @return [Array] An array of lowercased, alphabetized, Vet360 exception keys
+      #
+      def exception_keys
+        exceptions_file
+          .dig('en', 'common', 'exceptions')
+          .keys
+          .select { |exception| exception.include? 'VET360_' }
+          .sort
+          .map(&:downcase)
+      end
+
+      # Triggers the associated StatsD.increment method for the Vet360 buckets that are
+      # initialized in the config/initializers/statsd.rb file.
+      #
+      # @param *args [String] A variable number of string arguments. Each one represents
+      #   a bucket in StatsD.  For example passing in ('policy', 'success') would increment
+      #   the 'api.vet360.policy.success' bucket
+      #
+      def increment(*args)
+        buckets = args.map(&:downcase).join('.')
+
+        StatsD.increment("#{Vet360::Service::STATSD_KEY_PREFIX}.#{buckets}")
+      end
+
+      # If the passed response contains a transaction status that is in one of the final
+      # success or failure states, it increments the associated StatsD bucket.
+      #
+      # @param response [FaradayObject] The raw response from the Faraday HTTP call
+      # @param bucket1 [String] The Vet360 bucket to increment.  This bucket must
+      #   already be initialized in config/initializers/statsd.rb.
+      # @return [Nil] Returns nil only if the passed transaction status is not a final status
+      #
+      def increment_transaction_results(response, bucket1 = 'posts_and_puts')
+        status = status_in(response)
+
+        return unless final_status?(status)
+
+        increment(bucket1, bucket_for(status))
+      end
+
+      private
+
+      def exceptions_file
+        config = Rails.root + 'config/locales/exceptions.en.yml'
+
+        YAML.load_file(config)
+      end
+
+      def status_in(response)
+        response&.body&.dig('tx_status')&.upcase
```

Comment:
@wyattwalter @hpjaj I have some minor concerns over this being used as the key for the prometheus time series. My gut tells me that introducing an error on the Vet360 that would have the affect of making this a highly dynamic value is unlikely, but technically we're relying on the behavior of their system and I don't know how hard the contract is for this interface.

Reply:
@jkassemi - You bring up a good point.  I'll add in some logic that ensures that the exception key in question is already represented in our codebase, to mitigate against there being any misalignment.  

There are a few different ways I can go about implementing this.  I'll get something in a PR for us to speak to.

Thanks!

Reply:
@jkassemi I don't believe this is being used as the key for the time-series is it? This value is being used to bucket the statuses into either success or failed, which is then the key that gets sent to prometheus. 

Do you mind double checking my logic however though?

Reply:
@jkassemi - Here is the follow up PR to address a key being passed to prometheus that has not been accounted for:

https://github.com/department-of-veterans-affairs/vets-api/pull/2050